src/ipcl_python/ipcl_python.py

```diff
@@ -421,7 +414,9 @@ def __rsub__(
         self,
         other: Union["PaillierEncryptedNumber", np.ndarray, list, int, float],
     ) -> "PaillierEncryptedNumber":
-        return self.__sub__(other)
+        if isinstance(other, PaillierEncryptedNumber):
+            return other.__sub__(self)
+        return self.__mul__(-1.0).__raw_add(other)
```

Comment:
You shouldn't need to call the dunder methods in normal syntax, but use the symbols and built-in functions instead. That is the advantage of defining the dudner methods (a.k.a. operator overloading :))